source/jobs/JobEngine.cpp

```diff
@@ -153,6 +153,22 @@ void JobEngine::exec_action(PlainJobDocument::JobAction action, const std::strin
             return;
         }
     }
+    else if (action.type == "runShellCommand")
+    {
+        try
+        {
+            // build command for shell command
+            command = buildCommand(jobHandlerDir, "shell-command-handler.sh", jobHandlerDir);
```

Comment:
Another issue I see is we are hard coding the handler name here. This means user cannot have (1)handler with different name (2)multiple shell command handler.

Reply:
it is not an issue because shell-command-handler.sh is treated as a standard job handler and is created by us under sample-job-handler directory, whatever the shell-command job document will be, device client will invoke this handler to process